### ISSUE TITLE
Get IPsec vpn phase1 Pass

### ISSUE THREAD

**Issue by magarjoba** [FortiPower/PowerFGT#139]
Good day.
I am interested in the ability to extract the password from the phase in the tunnel. even if it is encrypted
in the form of such 
psksecret ENC ppDZlb3tJGNOcQqwi7O8ER/eOA2iYkzgfxWtDVkOZPnctdgBrJ7+F/QhwYnYt+J6YR

```
$mysecpassword = ConvertTo-SecureString "testtss" -AsPlainText -Force
#$fw3 = Connect-FGT 192.168.1.99 -Username admin -Password $mysecpassword -SkipCertificateCheck -DefaultConnection:$false
<#          Splatting 
-name     ,  If I want to add more than one interface name, I had to get out like this.
                      #>
$names = @(@{nam ="tunnel1"},@{nam ="tunnel2"},@{nam ="tunnel3"},@{nam ="tunnel4"})
foreach ($nam in $names) {
Get-FGTVpnIpsecPhase1Interface -connection $fw3  @nam -filter_type contains | Select mode, name, psksecret 
}

```

mode          name     psksecret
----              ----     ---------
aggressive tunnel1 ENC XXXX 
aggressive tunnel2 ENC XXXX 
aggressive tunnel3 ENC XXXX 
aggressive tunnel4 ENC XXXX 

As a result, I get passwords at the exit
))) XXXX !!! what ...
leafed through the api, I could not figure out how to pull the values. in its normal form. without decryption .. it is not important to me ..

I ask you to suggest any options. yes via ssh shows. why then does not work here.

**Comment by alagoutte** [comment]
Hi,

Actually, it is not possible but if remember, there is a tips to get the password not encrypted

**Comment by magarjoba** [comment]
> Hi,
> 
> Actually, it is not possible but if remember, there is a tips to get the password not encrypted

it's  work))
        }

        $response = Invoke-FGTRestMethod -uri 'api/v2/cmdb/vpn.ipsec/phase1-interface?plain-text-password=1' -method 'GET' -connection $connection @invokeParams
        $response.results
    }

**Comment by magarjoba** [comment]
> Hi,
> 
> Actually, it is not possible but if remember, there is a tips to get the password not encrypted

I'm thinking, so that I don't change the Get-FGTVpnIpsecPhase1Interface Function itself
How can I add this parameter when accessing in the code itself?
<#? plain-text-password = 1 #> ??
I haven't tested yet. but as I understand it, when unloading the WAN interfaces, if there is a ppoe connection method, then I will not see the password either. ?

**Comment by alagoutte** [comment]
> > Hi,
> > Actually, it is not possible but if remember, there is a tips to get the password not encrypted
> 
> it's work))
> }
> 
> ```
>     $response = Invoke-FGTRestMethod -uri 'api/v2/cmdb/vpn.ipsec/phase1-interface?plain-text-password=1' -method 'GET' -connection $connection @invokeParams
>     $response.results
> }
> ```

Yes, it is this tips, i will add to Get-FGTVpnIpsecPhase1Interface

**Comment by alagoutte** [comment]
> > Hi,
> > Actually, it is not possible but if remember, there is a tips to get the password not encrypted
> 
> I'm thinking, so that I don't change the Get-FGTVpnIpsecPhase1Interface Function itself
> How can I add this parameter when accessing in the code itself?
> <#? plain-text-password = 1 #> ??
it is not possible without modified Get-FGTVpnIpsecPhase1Interface function...

> I haven't tested yet. but as I understand it, when unloading the WAN interfaces, if there is a ppoe connection method, then I will not see the password either. ?
Good question, need to look where there is pppoe password (on Get-FgtSystemInterface ?), it is may be possible to use the same tips

**Comment by alagoutte** [comment]
Hi, 

I push a fix to add support of plaintext_password (See #140), it will be available on next major release...
